cloudinit/distros/networking.py

```diff
@@ -102,10 +106,71 @@ def is_vlan(self, devname: DeviceName) -> bool:
     def master_is_bridge_or_bond(self, devname: DeviceName) -> bool:
         return net.master_is_bridge_or_bond(devname)
 
+    @abc.abstractmethod
+    def settle(self, *, exists=None) -> None:
+        """Wait for device population in the system to complete.
+
+        :param exists:
```

Comment:
My reading of this parameter name made me think initially this was a boolean. This is probably just a poorly named parameter (could have been device_name). 

For this PR, can we please add typing hints to this exists parameter definition and maybe make to docstring for :param exists mention first that it's a Optional string of a specific DeviceName. If given, only perform.... 

Would you be opposed to me providing a separate PR to rename that param in udevadm_settle and settle from exists to device_name? There is only one callsite that uses the "exists" param in DataSourceAltCloud.py.